apps/api/src/app/tenant/dtos/update-tenant-request.dto.ts

```diff
@@ -0,0 +1,13 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { TenantCustomData } from '@novu/shared';
+
+export class UpdateTenantRequestDto {
+  @ApiProperty()
+  identifier?: string;
+
+  @ApiProperty()
+  name?: string;
+
+  @ApiProperty()
+  data?: TenantCustomData;
```

Comment:
all these are optional? if yes then use `@ApiPropertyOptional`

Reply:
and please provide the types on the `@ApiPropertyOptional`, check the `apps/api/src/app/topics/dtos/filter-topics.dto.ts` as an example

Reply:
> 

Yes the user can decide what he wants to update,

Reply:
> and please provide the types on the `@ApiPropertyOptional`, check the `apps/api/src/app/topics/dtos/filter-topics.dto.ts` as an example

I was not sure how to type `data: TenantCustomData` i thought maybe converting it to class or about something like:
  {
    type: Record,
    additionalProperties: {
      oneOf: [
        // { type: 'string' },
        { type: 'number' },
        { type: 'boolean' },
        { type: 'undefined' },
        {
          type: 'array',
          items: {
            type: 'string',
          },
        },
      ],
    },
  }
  
  are you aware of any good ways to define it?

Reply:
we don't require any specific shape restrictions for this field apart from it should be an object, so I would say that what you have rn is ok